pkg/issuer/acme/dns/dns.go

```diff
@@ -292,6 +292,30 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1.GenericIssuer
 		}
 	case providerConfig.Route53 != nil:
 		dbg.Info("preparing to create Route53 provider")
+
+		// For route53, you must specify either an AccessKeyID or a secret reference to an AccessKeyID
+		// but not both.
+		if providerConfig.Route53.AccessKeyID != "" && providerConfig.Route53.SecretAccessKeyID.Name != "" {
+			return nil, nil, fmt.Errorf("route53 accessKeyID and accessKeyIDSecretRef cannot both be specified")
+		}
+
+		// Default to the AccessKeyID literal in the configuration
+		secretAccessKeyID := strings.TrimSpace(providerConfig.Route53.AccessKeyID)
+
+		// If the AccessKeyID secret reference option is defined, override the secretAccessKeyID variable
```

Comment:
We shouldn't ever allow having both specified, so perhaps this code/text can be updated to reflect that :) (I think returning an error is valid given this isn't really an expected state to be in once the validation is done in the webhook/validation code)